rct229/rules/section5/section5rule31.py

```diff
@@ -0,0 +1,111 @@
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.utils.assertions import getattr_, assert_
+from rct229.utils.jsonpath_utils import find_all
+from rct229.utils.match_lists import match_lists_by_id
+
+
+MANUAL_CHECK_MSG = "SURFACE IN P-RMR HAS SUBSURFACES MODELED WITH DIFFERENT MANUAL SHADE STATUS. VERIFY IF SUBSURFACES MANUAL SHADE STATUS IN B-RMR ARE MODELED THE SAME AS IN P-RMR"
+
+class Section5Rule31(RuleDefinitionListIndexedBase):
+    """Rule 5 of ASHRAE 90.1-2019 Appendix G Section 5 (Envelope)"""
+
+    def __init__(self):
+        super(Section5Rule31, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, True),
+            each_rule=Section5Rule31.BuildingRule(),
+            index_rmr="baseline",
+            id="5-31",
+            description="Manual fenestration shading devices, such as blinds or shades, shall be modeled or not modeled the same as in the baseline building design.",
+            list_path="ruleset_model_instances[0].buildings[*]",
+        )
+
+    class BuildingRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section5Rule31.BuildingRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, True),
+                # Make sure surfaces are matched in SurfaceRule
+                list_path="surfaces[*]",
+                each_rule=Section5Rule31.BuildingRule.SurfaceRule(),
+                index_rmr="baseline",
+            )
+
+        class SurfaceRule(RuleDefinitionListIndexedBase):
+            def __init__(self):
+                super(Section5Rule31.BuildingRule.SurfaceRule, self).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, True, True),
+                    # Make sure subsurfaces are matched
+                    # List_path will be evaluated after manual check
+                    each_rule=Section5Rule31.BuildingRule.SurfaceRule.SubsurfaceRule(),
+                    index_rmr="baseline"
+                )
+
+            def is_applicable(self, context, data=None):
+                surface_b = context.baseline
+                surface_p = context.proposed
+                # Both are none, not applicable.
+                # The logic indicates either or both surface_b and surface_p have subsurfaces to be applicable
+                # But XOR could fails the test (b != p) - the equality is not checked in this rule
+                # b != p will be captured by data error exception in this test.
+                return (surface_b.get("subsurfaces") is None) and (surface_p.get("subsurfaces") is None)
+
+            def manual_check_required(self, context, calc_vals=None, data=None):
+                surface_p = context.proposed
+                # raise data error exception if proposed surface has no subsurfaces
+                subsurfaces_p = assert_(surface_p.get("subsurfaces"), f"No subsurfaces found in the proposed surface {surface_p['id']}")
```

Comment:
`assert_` has no return value, so this will not work as expected.

FYI:  jsonpath has a slick way of getting items that have a specified value. For example, to get all the subsurfaces with `has_manual_interior_shades` set to True, you can use:
```
subsurfaces_with_manual_interior_shades_p = find_all("subsurfaces[?(@.has_manual_interior_shades=True)]", surface_p)
```
If you also use jsonpath the get all the subsurfaces with 
```
subsurfaces_p = findall("subsurfaces[*]", surface_p)
```
Then you can simply check that
```
len(subsurfaces_with_manual_interior_shades_p) != 0 and len(subsurfaces_with_manual_interior_shades_p) != len(subsurfaces_p)
```
If surface_p has no subsurfaces, then the findall will return [] for both. In that case, the manual check test will fail. That makes sense to me – no manual check is required if a surface has no subsurfaces.